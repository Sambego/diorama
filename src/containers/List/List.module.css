.list {
  padding: 0;
  margin: 0;
  list-style: none;
}

.ol {
  margin: 0;
  max-width: 80vw;

  text-align: left;
}

.item {
  padding: 2rem 0;
  margin: 0;

  font-size: 3vw;
}
