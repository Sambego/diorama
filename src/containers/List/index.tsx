import React, { Children, cloneElement } from "react";
import PropTypes from "prop-types";

import styles from "./List.module.css";

type ListProps = {
	className?: string;
	style?: React.CSSProperties;
	children: React.ReactNode;
	ordered?: boolean;
};

function List({ children, ordered, style, className }: ListProps) {
	const renderItems = () => {
		if (Array.isArray(children)) {
			return Children.map(children, (child, index) => {
				if (React.isValidElement(child)) {
					return cloneElement(child, {
						// @ts-ignore
						className: `${styles.item} diorama-list-item`,
						// eslint-disable-next-line react/no-array-index-key
						key: index,
					});
				}
				return null;
			});
		}
		if (React.isValidElement(children)) {
			return cloneElement(children, {
				className: `${styles.item} diorama-list-item`,
			});
		}
		return null;
	};

	if (ordered) {
		return (
			<ol
				className={`${styles.ol} diorama-list diorama-list-unordered ${className}`}
				style={style}
			>
				{renderItems()}
			</ol>
		);
	}

	return (
		<ul
			className={`${styles.list} diorama-list diorama-list-unordered ${className}`}
			style={style}
		>
			{renderItems()}
		</ul>
	);
}

List.propTypes = {
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
	ordered: PropTypes.bool,
	style: PropTypes.shape({}),
};

List.defaultProps = {
	className: "",
	ordered: false,
	style: {},
};

export default List;
