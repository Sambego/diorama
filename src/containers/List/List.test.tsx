import React from "react";
import renderer from "react-test-renderer";
import List from ".";

it("renders correctly", () => {
	const tree = renderer
		.create(
			<List>
				<li>one</li>
				<li>two</li>
			</List>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
