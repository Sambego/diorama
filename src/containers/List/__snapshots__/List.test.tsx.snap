// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<ul
  className="_list_15h4c_1 diorama-list diorama-list-unordered "
  style={{}}
>
  <li
    className="_item_15h4c_14 diorama-list-item"
  >
    one
  </li>
  <li
    className="_item_15h4c_14 diorama-list-item"
  >
    two
  </li>
</ul>
`;
